src/core/crypto/ecdsa.cpp

```diff
@@ -55,162 +53,22 @@ namespace Ecdsa {
 
 Error P256::KeyPair::Generate(void)
 {
-    mbedtls_pk_context pk;
-    int                ret;
-
-    mbedtls_pk_init(&pk);
-
-    ret = mbedtls_pk_setup(&pk, mbedtls_pk_info_from_type(MBEDTLS_PK_ECKEY));
-    VerifyOrExit(ret == 0);
-
-    ret = mbedtls_ecp_gen_key(MBEDTLS_ECP_DP_SECP256R1, mbedtls_pk_ec(pk), MbedTls::CryptoSecurePrng, nullptr);
-    VerifyOrExit(ret == 0);
-
-    ret = mbedtls_pk_write_key_der(&pk, mDerBytes, sizeof(mDerBytes));
-    VerifyOrExit(ret > 0);
-
-    mDerLength = static_cast<uint8_t>(ret);
-
-    memmove(mDerBytes, mDerBytes + sizeof(mDerBytes) - mDerLength, mDerLength);
-
-exit:
-    mbedtls_pk_free(&pk);
-
-    return (ret >= 0) ? kErrorNone : MbedTls::MapError(ret);
-}
-
-Error P256::KeyPair::Parse(void *aContext) const
-{
-    Error               error = kErrorNone;
-    mbedtls_pk_context *pk    = reinterpret_cast<mbedtls_pk_context *>(aContext);
-
-    mbedtls_pk_init(pk);
-
-    VerifyOrExit(mbedtls_pk_setup(pk, mbedtls_pk_info_from_type(MBEDTLS_PK_ECKEY)) == 0, error = kErrorFailed);
-#if (MBEDTLS_VERSION_NUMBER >= 0x03000000)
-    VerifyOrExit(mbedtls_pk_parse_key(pk, mDerBytes, mDerLength, nullptr, 0, MbedTls::CryptoSecurePrng, nullptr) == 0,
-                 error = kErrorParse);
-#else
-    VerifyOrExit(mbedtls_pk_parse_key(pk, mDerBytes, mDerLength, nullptr, 0) == 0, error = kErrorParse);
-#endif
-
-exit:
-    return error;
+    return otPlatCryptoEcdsaGenerateKey(this);
```

Comment:
nit: given that these method implementations are single line, should we inline by moving to hpp file?